src/Service/Resolvers/CosmosQueryStructure.cs

```diff
@@ -121,7 +120,10 @@ private void Init(IDictionary<string, object?> queryParams)
             else
             {
                 Columns.AddRange(GenerateQueryColumns(selection.SyntaxNode.SelectionSet!, _context.Document, SourceAlias));
-                string entityName = MetadataProvider.GetEntityName(underlyingType.Name);
+                string typeName = GraphQLUtils.TryExtractGraphQLFieldModelName(underlyingType.Directives, out string? modelName) ?
+                    modelName :
+                    underlyingType.Name;
+                string entityName = MetadataProvider.GetEntityName(typeName);
```

Comment:
Need @tarazou9 to take a look to validate this change.

Reply:
Seems pretty straightforward to handle the specifics of cosmos here where entity name <==> Object type name in Graphql schema OR if differing, get the value declard in the `@model` directive.